protocols/2a4397/ngs_library_prep.ot2.py

```diff
@@ -0,0 +1,216 @@
+from opentrons import labware, instruments, modules, robot
+from otcustomizers import StringSelection
+
+metadata = {
+    'protocolName': 'Oxford Nanopore 16S Barcoding NGS Library Prep',
+    'author': 'Nick <protocols@opentrons.com>',
+    'source': 'Custom Protocol Request'
+}
+
+# create custom labware
+res12_name = 'axygen_12_reservoir_290ml'
+if res12_name not in labware.list():
+    labware.create(
+        res12_name,
+        grid=(12, 1),
+        spacing=(9, 0),
+        diameter=8.3,
+        depth=43,
+        volume=22000
+    )
+
+# load modules and labware
+rxn_plate = labware.load(
+    'biorad_96_wellplate_200ul_pcr', '1', 'sample plate')
+barcode_plate = labware.load(
+    'biorad_96_wellplate_200ul_pcr', '2', 'barcode plate')
+reagent_res = labware.load(
+    res12_name, '3', 'reagent reservoir')
+magdeck = modules.load('magdeck', '4')
+magplate = labware.load(
+    'biorad_96_wellplate_200ul_pcr',
+    '4',
+    'sample plate (on magnetic module)',
+    share=True
+)
+original_sample_plate = labware.load(
+    'biorad_96_wellplate_200ul_pcr', '5', 'original samples')
+reagent_rack = labware.load(
+    'opentrons_24_tuberack_eppendorf_1.5ml_safelock_snapcap', '6')
+tips10 = [
+    labware.load('opentrons_96_tiprack_10ul', slot) for slot in ['8', '9']]
+tips300 = [labware.load('opentrons_96_tiprack_300ul', slot) for slot in ['10']]
+
+# reagents
+nuc_free_water = reagent_rack.wells('A1')
+longamp_mm = reagent_rack.wells('B1')
+tris_hcl = reagent_rack.wells('A2')
+rap = reagent_rack.wells('B2')
+
+ampure_xp_beads = reagent_res.wells(0)
+etoh = reagent_res.wells()[1:3]
+liquid_trash = reagent_res.wells()[10:]
+
+
+def run_custom_protocol(
+        number_of_samples: int = 96,
+        barcode_start_well: str = 'A1',
+        p10_mount: StringSelection('right', 'left') = 'right',
+        p300_mount: StringSelection('left', 'right') = 'left'
+):
+    # checks
+    if number_of_samples > 96 or number_of_samples < 1:
+        raise Exception('Invalid number of samples.')
+
+    all_wells = [r + str(c) for r in 'ABCDEFGH' for c in range(1, 13)]
```

Comment:
Just a suggestion here, you can use:
```
if barcode_start_well not in barcode_plate.children_by_name:
  print(..)
```
`.children_by_name` returns a list of tuple of (well_name, well)

This adds more compatibility with other labware types, this can allow a change in labware without breaking the protocol

Reply:
good note. thanks